pandas/tests/io/test_parquet.py

```diff
@@ -368,6 +368,41 @@ def test_multiindex_with_columns(self, pa_ge_070):
             check_round_trip(df, engine, read_kwargs={'columns': ['A', 'B']},
                              expected=df[['A', 'B']])
 
+    def test_write_ignoring_index(self, engine):
+        # ENH 20768
+        # Ensure index=False omits the index from the written Parquet file.
+        df = pd.DataFrame({'a': [1, 2, 3], 'b': ['q', 'r', 's']})
+
+        # Because we're dropping the index, we expect the loaded dataframe to
+        # have the default integer index.
+        expected = df.reset_index(drop=True)
+
+        check_round_trip(df, engine, write_kwargs={'index': False},
+                         expected=expected)
+
+    def test_write_ignoring_custom_index(self, engine):
+        # ENH 20768
+        # Ensure index=False omits the index from the written Parquet file,
+        # even if we're using a custom one.s
+        df = pd.DataFrame({'a': [1, 2, 3], 'b': ['q', 'r', 's']},
+                          index=['zyx', 'wvu', 'tsr'])
+
+        expected = df.reset_index(drop=True)
+        check_round_trip(df, engine, write_kwargs={'index': False},
+                         expected=expected)
+
+    def test_write_ignoring_multiindex(self, engine):
+        # ENH 20768
+        # Ensure index=False omits multiindexes as well.
+        arrays = [['bar', 'bar', 'baz', 'baz', 'foo', 'foo', 'qux', 'qux'],
+                  ['one', 'two', 'one', 'two', 'one', 'two', 'one', 'two']]
+        df = pd.DataFrame({'one': [i for i in range(8)],
+                           'two': [-i for i in range(8)]}, index=arrays)
+
+        expected = df.reset_index(drop=True)
+        check_round_trip(df, engine, write_kwargs={'index': False},
+                         expected=expected)
```

Comment:
Those tests with a custom index are failing, but those are actual test failures you need to solve. Currently we raise an error for older pyarrow versions if there is a non-default index (because older pyarrow could not handle that). But when dropping the index, this error is not needed. So the code that checks this can be updated (see https://github.com/pandas-dev/pandas/blob/ad19d73c0c929add3b127bc3f0a81b509de44ca5/pandas/io/parquet.py#L141)

Reply:
I _think_ I fixed it correctly.